packages/blind_pool/examples/future_polling_with_pooled_mut.rs

```diff
@@ -0,0 +1,152 @@
+//! This example demonstrates how to use `insert_mut()` to store and poll a Future
+//! that requires mutable access through `Pin<&mut Self>`.
+//!
+//! Many async operations return futures that need to be polled with a mutable reference.
+//! The `PooledMut` type allows us to store such futures in a pool and poll them directly.
+
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll, Waker};
+
+use blind_pool::BlindPool;
+
+/// A simple future that completes after being polled a specific number of times.
+struct CountdownFuture {
+    remaining_polls: u32,
+}
+
+impl CountdownFuture {
+    fn new(poll_count: u32) -> Self {
+        Self {
+            remaining_polls: poll_count,
+        }
+    }
+}
+
+impl Future for CountdownFuture {
+    type Output = String;
+
+    fn poll(mut self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Self::Output> {
+        if self.remaining_polls == 0 {
+            Poll::Ready("Countdown completed!".to_string())
+        } else {
+            println!("Countdown: {remaining_polls} polls remaining", remaining_polls = self.remaining_polls);
+            self.remaining_polls = self.remaining_polls.saturating_sub(1);
+            Poll::Pending
+        }
+    }
+}
+
+/// A dummy waker for this example.
+struct DummyWaker;
+
+impl std::task::Wake for DummyWaker {
+    fn wake(self: std::sync::Arc<Self>) {
+        // In a real async runtime, this would wake up the executor
+    }
+}
+
+fn main() {
+    println!("Demonstrating PooledMut with Future polling");
+    println!("==========================================\n");
+
+    // Create a blind pool
+    let pool = BlindPool::new();
+
+    // Insert a future using insert_mut - this gives us exclusive mutable access
+    let mut future_handle = pool.insert_mut(CountdownFuture::new(3));
+
+    // Create a dummy waker and context for polling
+    let waker = Waker::from(std::sync::Arc::new(DummyWaker));
+    let mut context = Context::from_waker(&waker);
+
+    println!("Starting to poll the future...\n");
+
+    // Poll the future until completion
+    // We can do this because PooledMut implements DerefMut, giving us &mut access
+    loop {
+        // Pin the mutable reference to poll the future
+        let pinned_future = Pin::new(&mut *future_handle);
```

Comment:
Dereferencing a `Pooled` or `PooledMut` should already return a pinned reference, as pooled objects are always pinned. The lack of `Pin` is also a defect in the existing `Deref` implementation for `Pooled` - let's also fix that while we are at it.